sdk/cosmosdb/cosmos/test/internal/unit/hybridExecutionContext.spec.ts

```diff
@@ -0,0 +1,306 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import {
+  ClientConfigDiagnostic,
+  ClientContext,
+  ConsistencyLevel,
+  Constants,
+  CosmosClientOptions,
+  CosmosDbDiagnosticLevel,
+  DiagnosticNodeInternal,
+  FeedOptions,
+  GlobalEndpointManager,
+  QueryInfo,
+  RequestOptions,
+  DiagnosticNodeType,
+} from "../../../src";
+import { expect } from "chai";
+import {
+  HybridQueryExecutionContext,
+  HybridQueryExecutionContextBaseStates,
+} from "../../../src/queryExecutionContext/hybridQueryExecutionContext";
+import { HybridSearchQueryInfo } from "../../../src/request/ErrorResponse";
+import { GlobalStatistics } from "../../../src/request/globalStatistics";
+import assert from "assert";
+import { HybridSearchQueryResult } from "../../../src/request/hybridSearchQueryResult";
+import sinon from "sinon";
+
+function createTestClientContext(
+  options: Partial<CosmosClientOptions>,
+  diagnosticLevel: CosmosDbDiagnosticLevel,
+) {
+  const clientOps: CosmosClientOptions = {
+    endpoint: "",
+    connectionPolicy: {
+      enableEndpointDiscovery: false,
+      preferredLocations: ["https://localhhost"],
+    },
+    ...options,
+  };
+  const globalEndpointManager = new GlobalEndpointManager(
+    clientOps,
+    async (diagnosticNode: DiagnosticNodeInternal, opts: RequestOptions) => {
+      expect(opts).to.exist; // eslint-disable-line no-unused-expressions
+      const dummyAccount: any = diagnosticNode;
+      return dummyAccount;
+    },
+  );
+  const clientConfig: ClientConfigDiagnostic = {
+    endpoint: "",
+    resourceTokensConfigured: true,
+    tokenProviderConfigured: true,
+    aadCredentialsConfigured: true,
+    connectionPolicyConfigured: true,
+    consistencyLevel: ConsistencyLevel.BoundedStaleness,
+    defaultHeaders: {},
+    agentConfigured: true,
+    userAgentSuffix: "",
+    pluginsConfigured: true,
+    sDKVersion: Constants.SDKVersion,
+    ...options,
+  };
+  const clientContext = new ClientContext(
```

Comment:
We already have an test/public/common/MockClientContext.ts can we reuse it here?